tests/timeseries/test_logical_meter.py

```diff
@@ -111,6 +116,147 @@ async def test_grid_power_2(
         assert len(results) == 10
         assert equal_float_lists(results, meter_sums)
 
+    async def test_grid_frequency_1(self, mocker: MockerFixture) -> None:
+        """Test the grid frequency formula."""
+        mockgrid = MockMicrogrid(grid_meter=True)
+        mockgrid.add_batteries(2)
+        mockgrid.add_solar_inverters(1)
+        await mockgrid.start(mocker)
+
+        grid_freq = microgrid.frequency()
+        grid_freq_recv = grid_freq.new_receiver()
+
+        assert grid_freq._task is not None
+        # We have to wait for the metric request to be sent
+        await grid_freq._task
```

Comment:
But I think this hack for syncing is also perfectly fine, you are only accessing the implementation of what you are testing.